OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -0,0 +1,1055 @@
+                    arrayRef.asVar, context, arrayDepender, stmts, _ eq ObjectType.Field,
+                    () => failure(accessPC, actualReceiver, actualParameter, baseMatchers)
+                ) { (allocationContext, allocationIndex, stmts) =>
+                        matchers ++= handleGetField(
+                            allocationContext, accessPC, allocationIndex,
+                            actualReceiver, actualParameter,
+                            baseMatchers, stmts
+                        )
+                    }
+
+            case _ =>
+                if (HighSoundnessMode) {
+                    matchers += AllFieldsMatcher
+                } else {
+                    indirectFieldAccesses.addIncompleteAccessSite(accessPC)
+                    matchers += NoFieldsMatcher
+                }
+        }
+
+        matchers
+    }
+}
+
+class FieldGetAnalysis private[analyses] (
+        final val project:       SomeProject,
+        final val apiMethodName: String,
+        final val fieldType:     Option[BaseType] = None
+) extends ReflectionAnalysis with FieldInstanceBasedReflectiveFieldAccessAnalysis {
+
+    override val apiMethod: DeclaredMethod = declaredMethods(
+        ObjectType.Field,
+        "",
+        ObjectType.Field,
+        apiMethodName,
+        MethodDescriptor.apply(ObjectType.Object, fieldType.getOrElse(ObjectType.Object))
+    )
+
+    protected def handleFieldAccess(
+        accessContext:      ContextType,
+        accessPC:           Int,
+        fieldVar:           V,
+        fieldGetParameters: Seq[Option[Expr[V]]],
+        stmts:              Array[Stmt[V]]
+    )(implicit state: ReflectionState[ContextType], indirectFieldAccesses: IndirectFieldAccesses): Unit = {
+        val fieldGetReceiver: Option[V] = fieldGetParameters.head.map(_.asVar)
+
+        val baseMatchers = Set(
+            MatcherUtil.retrieveSuitableMatcher[V](
+                fieldGetReceiver,
+                accessPC,
+                v => new ActualReceiverBasedFieldMatcher(v.value.asReferenceValue)
+            ),
+            MatcherUtil.retrieveSuitableNonEssentialMatcher[BaseType](
+                fieldType,
+                v => new UBTypeBasedFieldMatcher(v)
+            )
+        )
+
+        val persistentReceiver = fieldGetReceiver.flatMap(r => persistentUVar(r)(stmts))
+        val depender = FieldDepender(accessPC, persistentReceiver, None, baseMatchers)
+
+        AllocationsUtil.handleAllocations(
+            fieldVar, accessContext, depender, state.tac.stmts, _ eq ObjectType.Field,
+            () => failure(accessPC, persistentReceiver, None, baseMatchers)
+        ) { (allocationContext, allocationIndex, stmts) =>
+                val allMatchers = handleGetField(
+                    allocationContext, accessPC, allocationIndex,
+                    persistentReceiver, None,
+                    baseMatchers, stmts
+                )
+                addFieldRead(
```

Comment:
Probably fits in a single line

Reply:
Fun fact: Its counterpart (in the set analysis) also does exactly!